system-addon/lib/HighlightsFeed.jsm

```diff
@@ -141,13 +152,10 @@ this.HighlightsFeed = class HighlightsFeed {
       case at.INIT:
         this.init();
         break;
-      case at.NEW_TAB_LOAD:
-        if (this.highlightsLength < HIGHLIGHTS_MAX_LENGTH) {
-          // If we haven't filled the highlights grid yet, fetch again.
-          this.fetchHighlights(true);
-        } else if (Date.now() - this.highlightsLastUpdated >= HIGHLIGHTS_UPDATE_TIME) {
+      case at.SYSTEM_TICK:
+        if (Date.now() - this.highlightsLastUpdated >= HIGHLIGHTS_UPDATE_TIME) {
```

Comment:
There's 3 different "timers" that are being considered here (and for top sites):
1) system tick interval (5 minutes)
2) feed last updated (15 minutes)
3) links cache expiration (5 minutes)

Does it make sense to still have all 3? Or maybe a related question is what's each of their purpose now?

Another issue with similar but slightly different timers is that if they're a little bit off, an expected 5 minute refresh gets skipped and is only actually refreshed after 10 minutes. E.g., due to async stuff, last update was at T+5.01 and the next system tick at T+10.0 is only 4.99 minutes after the last update.

Reply:
I think that HIGHLIGHTS_UPDATE_TIME is the only one not needed anymore. Looks like a work around to avoid frequent updates on every page load.

1. system tick interval is needed because this is the new event we react to, independent of page loads. This should also fix the problem of refreshing highlights before having the image ready [as described in comment](https://github.com/mozilla/activity-stream/pull/3695#issuecomment-336592534)

2. links cache expiration ensures we get new highlights regardless of special events (like BOOKMARKS_REMOVED/ADDED) that invalidate the cache or our different invalidation tactics (every LAST_FEED_UPDATED minutes).

I'm not sure the timing issue is very important (maybe I don't get it) but we force update on special events for blocking and bookmarking where we need to ensure consistent information. If we don't update the highlights with visits once in a while it should be ok.

Reply:
The "don't update the highlights with visits once in a while" is actually "every other time" if the timing is just off. It's just being clear that the code might say "refresh every 5 minutes" but turns out to be "refresh every 10 minutes." That may be acceptable, but if 5 minutes was picked for a particular reason, it shouldn't just accidentally be twice as long.

Reply:
Could we move 
```
this.highlightsLastUpdated = Date.now();
```
At the top of the `fetchHighlights` method? I've tried it and it seems to consistently trigger updates at the right moment.

Reply:
I thought you were suggesting HIGHLIGHTS_UPDATE_TIME wouldn't be necessary, so there shouldn't be a need to track `highlightsLastUpdated` ? We could probably just set LinksCache update time to be 10% less than SYSTEM_TICK, i.e., 4.5 minutes, if we're sure we just want to update at least once every 5 minutes.